suites/message/nested.go

```diff
@@ -33,61 +35,235 @@ func TestNestedSends(t *testing.T, factory state.Factories) {
+		_, err := td.State().Actor(newAddr)
+		assert.Error(t, err)
+	})
+
+	//
+	// The following tests exercise invalid semantics of the inner message
+	//
+
+	t.Run("fail nonexistent ID address", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+
+		newAddr := utils.NewIDAddr(t, 1234)
+		amtSent := abi.NewTokenAmount(1)
+		recpt := stage.send(newAddr, amtSent, builtin.MethodSend, nil, 1)
+		assert.Equal(t, exitcode.Ok, recpt.ExitCode)
+
+		td.AssertBalance(stage.msAddr, multisigBalance) // No change.
+		_, err := td.State().Actor(newAddr)
+		assert.Error(t, err)
+	})
+
+	t.Run("fail nonexistent actor address", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+
+		newAddr := utils.NewActorAddr(t, "1234")
+		amtSent := abi.NewTokenAmount(1)
+		recpt := stage.send(newAddr, amtSent, builtin.MethodSend, nil, 1)
+		assert.Equal(t, exitcode.Ok, recpt.ExitCode)
+
+		td.AssertBalance(stage.msAddr, multisigBalance) // No change.
+		_, err := td.State().Actor(newAddr)
+		assert.Error(t, err)
+	})
+
+	t.Run("fail invalid methodnum new actor", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
 
+		newAddr := td.Wallet().NewSECP256k1AccountAddress()
 		amtSent := abi.NewTokenAmount(1)
-		recpt := stage.send(aliceId, amtSent, builtin.MethodSend, nil, 1)
+		recpt := stage.send(newAddr, amtSent, abi.MethodNum(99), nil, 1)
 		assert.Equal(t, exitcode.Ok, recpt.ExitCode)
-		balanceAfter := td.GetBalance(aliceId)
 
-		assert.Equal(t, big.Sub(big.Add(balanceBefore, amtSent), recpt.GasUsed), balanceAfter)
+		td.AssertBalance(stage.msAddr, multisigBalance) // No change.
+		_, err := td.State().Actor(newAddr)
+		assert.Error(t, err)
+	})
+
+	t.Run("fail invalid methodnum for actor", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+		balanceBefore := td.GetBalance(stage.creator)
+
+		amtSent := abi.NewTokenAmount(1)
+		recpt := stage.send(stage.creator, amtSent, abi.MethodNum(99), nil, 1)
+		assert.Equal(t, exitcode.Ok, recpt.ExitCode)
+
+		td.AssertBalance(stage.msAddr, multisigBalance)                        // No change.
+		td.AssertBalance(stage.creator, big.Sub(balanceBefore, recpt.GasUsed)) // Pay gas, don't receive funds.
+	})
+
+	t.Run("fail non-CBOR params", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+
+		newAddr := td.Wallet().NewSECP256k1AccountAddress()
+		amtSent := abi.NewTokenAmount(1)
+		params := typegen.Deferred{Raw: []byte{1, 2, 3, 4}}
+		recpt := stage.send(newAddr, amtSent, builtin.MethodSend, &params, 1)
+		assert.Equal(t, exitcode.Ok, recpt.ExitCode)
```

Comment:
Asserting the execution is fine here (Which i agree with) seems to be at odds with asserting the target actor doesnt get created down below on line 230

Reply:
Ah, I think i get what youre asserting here. Invalid parameters should cause the internal send to fail.
I think that `MethodSend` should not care at all about the format of its parameters, the only reason we care that things are CBOR is because we happen to choose cbor as the encoding for our actors (an actor implementation should be able to choose whatever encoding they like, technically). And therefore, unless the receiver method tries to decode parameters, it shouldnt care whats in that field. (this can be used by payment processors to tag transactions, among other things)

Reply:
Ok